pkg/controller/signup.go

```diff
@@ -73,26 +119,25 @@ func (s *Signup) VerifyCodeHandler(ctx *gin.Context) {
 	}
 
 	userID := ctx.GetString(context.SubKey)
-	signupResource, err := s.signupService.GetUserSignup(userID)
-	if err != nil {
-		log.Error(ctx, err, "error getting UserSignup resource")
-		errors.AbortWithError(ctx, http.StatusInternalServerError, err, "error getting UserSignup resource")
-		return
-	}
 
-	if signupResource == nil {
-		log.Errorf(ctx, nil, "UserSignup resource for userID: %s resource not found", userID)
-		ctx.AbortWithStatus(http.StatusNotFound)
+	signup, err := s.signupService.GetUserSignup(userID)
+	if err != nil {
+		if errors2.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			errors.AbortWithError(ctx, http.StatusNotFound, err, "usersignup not found")
+		}
+		log.Error(ctx, err, "error retreiving usersignup")
+		errors.AbortWithError(ctx, http.StatusInternalServerError, err, fmt.Sprintf("error retreiving usersignup: %s", userID))
 		return
 	}
 
 	// The VerifyCode() call here MAY make changes to the specified signupResource
-	err = s.verificationService.VerifyCode(ctx, signupResource, code)
+	signup, err = s.verificationService.VerifyCode(signup, code)
 
 	// Regardless of whether the VerifyCode() call returns an error or not, we need to update the UserSignup instance
 	// as its state can be updated even in the case of an error.  This may result in the slight possibility that any
 	// errors returned by VerifyCode() are suppressed, as error handling for the UserSignup update is given precedence.
-	_, err2 := s.signupService.UpdateUserSignup(signupResource)
+	_, err2 := s.signupService.UpdateUserSignup(signup)
```

Comment:
There is no need for a second `err2` var: you could use `err` again here, couldn't you?

Reply:
so the reason why we are doing this is because we are temporarily ignoring the errors in err in favour of updating the usersignup. if the update fails we will return err2, if not and there was an err we will return that. It's described in this comment above: 
```
// Regardless of whether the VerifyCode() call returns an error or not, we need to update the UserSignup instance
// as its state can be updated even in the case of an error.  This may result in the slight possibility that any
// errors returned by VerifyCode() are suppressed, as error handling for the UserSignup update is given precedence.
```

i was thinking that what we could do instead is return a array of json errors, that way we aren't skipping or missing any errors that occur.. WDYT?